### PR TITLE
fix(absorb): honor declared links before generic mount fallback

### PR DESCRIPTION
**Pull request by yukimemi**
## Problem

`yui absorb <target>`'s reverse lookup (`find_source_for_target`) checked
generic `[[mount.entry]]` entries **before** `[[link]]`/`.yuilink`
declarations. The forward `apply` walk (`walk_and_link_body`) does the
opposite: it keeps recursing past a dir-scoped ancestor link so a nested
marker's override still fires. Result: `absorb` could propose a different
source path than `apply` would actually populate for the same target.

Concretely: with `home/.config` dir-junctioned to `~/.config`, and a nested
`home/.config/<app>/.yuilink` overriding `dst` to `%APPDATA%/<app>`,
`yui absorb` resolved the target back through the generic `home` -> `~`
mount instead of the more specific marker.

## Fix

- Swap the two search passes in `find_source_for_target` so declared
  `[[link]]`/`.yuilink` entries are checked first, generic mount entries
  are the fallback — matching `apply`'s precedence.
- Add `TargetMatch.specific: bool` to distinguish a declared-link match
  from a generic-mount-fallback match (needed by the feature below).

## New: `yui absorb --to <SRC>`

Filing an absorb for a target nothing claims yet used to require
hand-authoring a `.yuilink` marker first. `--to <SRC>` (a path relative to
`$DOTFILES`) now creates that marker on the fly:

- Templatizes the target's absolute path against known env-var roots
  (`LOCALAPPDATA`/`APPDATA`/`USERPROFILE` on Windows,
  `XDG_CONFIG_HOME`/`XDG_CACHE_HOME`/`XDG_DATA_HOME`/`HOME` elsewhere) for
  portability across machines — same convention as hand-written markers
  like `home/.config/shoka/.yuilink`.
- Only fires when nothing *specific* claims the target yet — a generic
  mount-fallback match is exactly what `--to` is meant to override. If a
  *different* specific declaration already claims the target, it bails
  instead of silently redirecting.
- Directory targets only (the realistic app-data-directory case); a file
  target errors with a hint to write the `[[link]]` by hand.

## Verification

- `cargo build`, `cargo fmt --check`, `cargo clippy --all-targets`: clean.
- `cargo test`: 252 passed, 0 failed (4 existing `absorb()` test callsites
  updated for the new `to` parameter).
- Manual smoke test against a throwaway fixture: `--to` creates the marker,
  templatizes via `LOCALAPPDATA`, absorbs directory content, and a
  subsequent `absorb --dry-run` (no `--to`) resolves through the new
  declaration.

## Summary by CodeRabbit

- **New Features**
  - Added a `--to <SRC>` option to `yui absorb <target>`.
  - When the target has no existing source declaration, `absorb` can create the source directory and add a portable `.yuilink` marker automatically.
  - The command refuses to proceed when another declaration already claims the target.

- **Documentation**
  - Updated the `yui absorb` command reference and anomaly-handling guidance with the new option and behavior.

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -106,8 +106,21 @@ pub enum Command {
     /// `--yes`, prompts on a TTY before writing; off a TTY refuses
     /// to act unless `--yes` is given. `--dry-run` only shows the
     /// diff and exits.
+    ///
+    /// If no `[[mount.entry]]` / `[[link]]` already claims `target`,
+    /// pass `--to` to declare where it should live in source: yui
+    /// creates that directory, writes a `.yuilink` marker mapping it
+    /// back to `target` (substituting a known env var like `%APPDATA%`
+    /// / `$XDG_CONFIG_HOME` for portability when the target falls
+    /// under one), and proceeds with the absorb.
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/cli.rs]: Fixed — updated the doc comment: --to may override a generic [[mount.entry]] fallback, and only bails on a conflicting *specific* [[link]]/.yuilink declaration. Also fixed the matching README wording (both the prose block and the command table). @coderabbitai

File: src/cmd/absorb.rs [hunk 2]
```diff
@@ -39,15 +42,45 @@ pub fn absorb(
...
-        None => anyhow::bail!(
+        specific: _,
+    } = match (found, to) {
+        (Some(m), None) => m,
+        (Some(m), Some(rel)) if !m.specific => {
+            // Only the generic mount-derived fallback claims this
+            // target — no explicit `[[link]]`/`.yuilink` declaration
+            // does. That's exactly what `--to` exists to override with
+            // a new, more specific declaration.
+            create_marker_for(&source, &rel, &target, &config.mount.marker_filename)?
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/cmd/absorb.rs]: Good catch — this was a real bug: --to wrote the directory + marker before the --dry-run check and before the y/N prompt, so --dry-run mutated the repo and a declined confirmation left the marker behind. Fixed by splitting create_marker_for into plan_marker_for (validates + renders content, no I/O) and write_marker (the actual fs::create_dir_all + fs::write), called only after both the dry-run short-circuit and the confirmation gate pass. Verified with a throwaway fixture: --dry-run and a declined prompt now leave zero trace; --yes still creates + absorbs correctly. @coderabbitai

File: src/cmd/absorb.rs [hunk 3]
```diff
@@ -108,6 +141,115 @@ pub fn absorb(
...
+             literal path, which won't portable across machines/users. \
+             Edit {marker_path} by hand to templatize it."
+        ),
+    }
+
+    std::fs::create_dir_all(&marker_dir)?;
+    std::fs::write(
+        &marker_path,
+        format!(
+            "[[link]]\ndst = \"{dst}\"\nwhen = \"yui.os == '{}'\"\n",
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/cmd/absorb.rs]: Fixed — dst is now serialized via toml::Value::String(dst).to_string() instead of hand-rolled "{dst}" interpolation, so quotes/backslashes/control chars in the templatized path can't produce an unparseable or misinterpreted marker. @coderabbitai

File: src/cmd/absorb.rs [hunk 4]
```diff
@@ -108,6 +141,115 @@ pub fn absorb(
...
+            continue;
+        };
+        let val = val.trim_end_matches(['/', '\\']);
+        if val.is_empty() {
+            continue;
+        }
+        let matches = if windows {
+            target_str.len() > val.len() && target_str[..val.len()].eq_ignore_ascii_case(val)
+        } else {
+            target_str.len() > val.len() && target_str.starts_with(val)
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/cmd/absorb.rs]: Fixed — templatize_dst now requires a path-separator boundary right after the matched root (target_str.as_bytes()[val.len()] must be '/' or '\\'), not just a string prefix. Also split the matching logic into templatize_dst_with(target, lookup) so it's unit-testable without mutating process-wide HOME/USERPROFILE/APPDATA (which other tests depend on for real) — added 3 tests including a regression test for exactly this boundary case (HOME=/home/alice must not swallow /home/alice2/app). @coderabbitai

[the remark this answers is not in the document]
Comment (yukimemi) [reply src/cmd/absorb.rs]: Fixed — allowed `target_str.len() >= val.len()` and treat an exact-root match (no trailing byte to check) as a valid boundary too. Added `matches_when_target_is_exactly_the_root` regression test alongside the existing boundary tests. @coderabbitai
